src/common/io/socket/address.c

```diff
@@ -0,0 +1,187 @@
+/***********************************************************************************************************************************
+Address Info
+***********************************************************************************************************************************/
+#include "build.auto.h"
+
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#include <sys/socket.h>
+
+#include "common/debug.h"
+#include "common/io/socket/address.h"
+#include "common/log.h"
+#include "common/memContext.h"
+
+/***********************************************************************************************************************************
+Object type
+***********************************************************************************************************************************/
+struct AddressInfo
+{
+    AddressInfoPub pub;                                             // Publicly accessible variables
+};
+
+/***********************************************************************************************************************************
+Free address list
+***********************************************************************************************************************************/
+static void
+addrInfoFreeResource(THIS_VOID)
+{
+    THIS(AddressInfo);
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(ADDRESS_INFO, this);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+
+    freeaddrinfo(*(struct addrinfo **)lstGet(this->pub.list, 0));
+
+    FUNCTION_LOG_RETURN_VOID();
+}
+
+/**********************************************************************************************************************************/
```

Comment:
Presumably could use a description ... along with a few others in here.

Reply:
Added more description of what is being freed here in 8a8e58c4b6a. extern'd functions have their comment in the header.